src/app/clients/fetchDataFromServer.ts

```diff
@@ -0,0 +1,60 @@
+import supabase from "@/supabaseServer";
+
+export type CongestionChargeDetails = {
+    postcode: string;
+    congestionCharge: boolean;
+};
+
+export const getCongestionChargeDetails = async (
+    processingData: ProcessingData
+): Promise<CongestionChargeDetails[]> => {
+    const postcodes = [];
+    for (const parcel of processingData) {
+        postcodes.push(parcel.client!.address_postcode);
+    }
+
+    const response = await supabase.functions.invoke("check-congestion-charge", {
+        body: { postcodes: postcodes },
+    });
+
+    return JSON.parse(response.data);
+};
+
+export type ProcessingData = Awaited<ReturnType<typeof getProcessingData>>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const getProcessingData = async () => {
+    const response = await supabase
```

Comment:
Do we not need some sort of error handling here?

Reply:
that's your ticket! you've already done a great job at it all
https://github.com/VauxhallFoodbank/vauxhall-foodbank/pull/70/files#diff-af4456595211b28ff643ecd6021325966085c08d00ffc74a5e3c0188c83c6c34R55-R60
this is the exact same line!